include/envoy/network/connection.h

```diff
@@ -96,6 +96,10 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
     Stats::Counter* bind_errors_;
     // Optional counter. Delayed close timeouts will not be tracked if this is nullptr.
     Stats::Counter* delayed_close_timeouts_;
+    // Optional histogram, used to track quic self-initiated connection close error if not nullptr.
+    Stats::Histogram* quic_connection_self_close_error_;
```

Comment:
I think these stats should be on a transport_socket or similar, because they don't apply to non-quic connections.

Reply:
source/common/http/http3/codec_stats.h might be a better place for them.